src/app/scopes-dialog.component.ts

```diff
@@ -16,21 +16,89 @@ declare let fabric, mwf;
 })
 export class ScopesDialogComponent extends GraphExplorerComponent implements AfterViewInit {
 
+  // Use this to get a handle on the scopes-list-table-container element.
+  @ViewChild('scopesListTableContainer') scopesTableList: ElementRef;
+
+  // Contains the scopes list table height. The maximum height value is 451px.
+  scopesListTableHeight: String;
+
+  // Flags for changing the scopes list table height.
+  hasChangedScopeListHeight: Boolean = false;
+  hasRequestedAdminConsent: Boolean = false;
+
+  // Updates the style.height of the scopes-list-table-container element. 
+  getScopesListTableHeight(): String {
+    this.scopesListTableHeight = window.getComputedStyle(this.scopesTableList.nativeElement, null).getPropertyValue("height");
+    return this.scopesListTableHeight;
+  }
+
   getScopeLabel(scopeName: String): String {
     return scopeName + " scope";
   }
 
   ngAfterViewInit(): void {
     ScopesDialogComponent.setScopesEnabledTarget();
-    window['launchPermissionsDialog'] = ScopesDialogComponent.showDialog
+    window['launchPermissionsDialog'] = ScopesDialogComponent.showDialog;
+    this.scopesListTableHeight = window.getComputedStyle(this.scopesTableList.nativeElement, null).getPropertyValue("height");
   }
 
+  /** 
+   * Indicates whether the scope list has been changed.
+   * @returns {boolean} A value of true indicates that a scope has been added or removed from the scope list.
+  */
   scopeListIsDirty(): boolean {
-    return PermissionScopes.filter((s) => s.enabled !== s.enabledTarget).length > 0;
+
+    // Determine whether the scope list has changed. The scope list has changed if isDirty = true.
+    let isDirty = PermissionScopes.filter((s) => s.enabled !== s.enabledTarget).length > 0;
+
+    // Reduce the size of the scopes table list by the size of the message bar. We only want to make this
+    // change the first time that the selected scope list is changed. 
+    if (isDirty && !this.hasChangedScopeListHeight) {
+
+      // Convert the table height from string to number.
+      var currentHeight: number = Number(this.scopesListTableHeight.replace(/px/, "")).valueOf();
+
+      // Update the table height based on the height of the message bar that's displayed when the scope list is dirty.
+      // This should keep the buttons from moving below the viewport as long as the component is showing in the viewport.
+      var updatedHeight: number = currentHeight - 60;
```

Comment:
Can we update "60" with the height of the message bar?

Reply:
Same answer as above.